### PR TITLE
Theme: Add new theme called "Base"

### PR DESCRIPTION
**Pull request by rxaviers**
Keep "Smoothness" and "UI Lightness" in the gallery, while the new
theme will be available as "Base" and it will be the default.

Ref https://github.com/jquery/jquery-ui/pull/1384

TODO:
- [x] Remove a temporary change below before landing.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -7,6 +7,11 @@
 			"stable": true
 		},
 		{
+			"version": "1a10f10cb0a39281e91eec5a53b5bb874497c629",
+			"dependsOn": "jQuery1.6+",
+			"label": "Experimental"
+		},
+		{
```
Comment (rxaviers) [inline config.json]: Note to self. This is a temporary change. Remove before landing.

File: themeroller.js [hunk 2]
```diff
@@ -82,6 +82,7 @@ Frontend.prototype = {
 
 	css: function( vars ) {
 		var theme = new ThemeRoller({
+			version: "1.12.0-pre",
```
Comment (rxaviers) [inline themeroller.js]: Note to self. This is a temporary change. Remove before landing.

**Comment by rxaviers** [comment]
Follow these instructions to run the server locally using the temporary changes:

1. Run `grunt prepare`. It will fail, because it won't find version `1a10f10cb0a39281e91eec5a53b5bb874497c629`.
1. Run `cd tmp/jquery-ui && git checkout --detach 20a636844961a1fb2de7a892ab28890091822e6a && git pull origin refs/pull/1384/head:pr/1384`.
1. Run `grunt prepare`. I must pass now.
1. Run `node server.js --console` and browse it.

**Comment by rxaviers** [comment]
Themeroller displaying the new theme called "Base".

screenshot from 2015-01-08 17 00 59

**Comment by rxaviers** [comment]
jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages with the new theme called "Base":
- https://gist.github.com/rxaviers/1f77d0a19806df75ff76

**Comment by kborchers** [comment]
Didn't I see a better mockup somewhere with the rounded corners removed and a better font? This feels very similar to smoothness with just a slight color and maybe a font change when I look at it.

**Comment by jzaefferer** [comment]
We changed the font-size in general, independent of this new theme. That doesn't seem to be included here, though. Rounded corners are still in the new theme, its just [smaller radii(?)](https://github.com/jquery/jquery-ui/pull/1384/files#diff-ff24a972bba1a2ecf79314fd9a5d048fR376). The smaller font-size might make the rounded corners more obvious.

One thing that looks very wrong on the screenshot is the overlay and shadow element demo. The overlay doesn't cover the background, the shadow also disappears in the middle.

**Comment by rxaviers** [comment]
> We changed the font-size in general, independent of this new theme.

Please, can you point me to this change?

**Comment by rxaviers** [comment]
> One thing that looks very wrong on the screenshot is the overlay and shadow element demo. The overlay doesn't cover the background, the shadow also disappears in the middle.

I'm assuming changes are going to be made on jquery/jquery-ui#1384, so I can later update this PR accordingly.

**Comment by jzaefferer** [comment]
> Please, can you point me to this change?

https://github.com/jquery/jquery-ui/pull/1374

**Comment by kborchers** [comment]
> Rounded corners are still in the new theme, its just smaller radii(?). The smaller font-size might make the rounded corners more obvious.

Let me rephrase my comment. Can we get rid of the rounded corners in this new base theme to make it feel more current?

**Comment by jzaefferer** [comment]
You can check out the actual theme and discuss it at the original PR: https://github.com/jquery/jquery-ui/pull/1374

**Comment by kborchers** [comment]
> You can check out the actual theme and discuss it at the original PR: jquery/jquery-ui#1374

I thought this was the Theme PR. The one you referenced is just the font size change so this would be the appropriate place to discuss the removal of the rounded corners. It doesn't seem like too much to ask for since most themes for current frameworks have removed rounded corners. They make it feel outdated no matter what you do to color or font.

**Comment by kborchers** [comment]
Maybe you meant https://github.com/jquery/jquery-ui/pull/1384 which I see now. I will comment there.

**Comment by jzaefferer** [comment]
Yes, sorry, that's the one I meant.

**Comment by rxaviers** [comment]
This PR is now (temporarily) pointing to d2eb0c93cfc5fce158ac2af6ca7b0f50760d9b90, i.e., @jzaefferer's https://github.com/jquery/jquery-ui/pull/1420 updates.

I have it deployed locally. Please, just let me know if I can provide a screenshot of a particular place or perform any test and post here.

**Comment by rxaviers** [comment]
Updated jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages using the new "Base" theme (based on d2eb0c93cfc5fce158ac2af6ca7b0f50760d9b90). They're available for download on https://gist.github.com/rxaviers/edd9cb1ecdedb3e65b10.

Below is how I see the tooltip dropshadow on 1.12.0-pre package and on 1.11.2.

| 1.12.0-pre | 1.11.2 |
| --- | --- |
| screenshot from 2015-01-21 09 49 59 | screenshot from 2015-01-21 09 50 03 |

The Overlay shadow also changed. It looks better on 1.12.0-pre (although, the white stripe underneath it is still present). Using the Base theme on 1.11.2 or older packages is still problematic.

| 1.12.0-pre | 1.11.2 |
| --- | --- |
| screenshot from 2015-01-21 09 36 03 copy | screenshot from 2015-01-21 09 35 59 |

**Comment by rxaviers** [comment]
I've noticed a problem that happens since my first update. The initial Icons on 1.12-pre package are all duplicate / the same. In contrast, this doesn't happen on 1.11.2 (or older) using the Base theme.

| 1.12.0-pre | 1.11.2 |
| --- | --- |
| screenshot from 2015-01-21 09 36 03 | screenshot from 2015-01-21 09 35 59 copy |

**Comment by rxaviers** [comment]
Jörn and I had a chat. Below is the outcome:
- [ ] Create a new `template/zip/index-1-12.html` with several style and markup fixes: (a) drop body font size override, (b) fix overlay markup, (c) fix caret icons (s/ui-icon-carat/ui-icon-caret/g), (d) fix menu markup (add div inside li's).
- [ ] Do not use an image when using the flat texture.
- [ ] Update "Base" theme variables according to @jzaefferer's tooltip shadow update.
- The white stripe of the overlay background is a local thing and should work just fine on the server (it'is due to bad image background generation).
- Should we change overlay background color.

Double check:
- [ ] On 1.9 and 1.10, accordion is not working properly in index.html demo page.

**Comment by rxaviers** [comment]
I've edited my previous comment.

**Comment by rxaviers** [comment]
Updated jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages using the new "Base" theme (based on a2db6209ca77c901b0195683d330d1d8c935bb12). They're available for download on https://gist.github.com/rxaviers/5293bc717f283f5eaad7

**Comment by rxaviers** [comment]
Updated jQuery UI 1.9.2, 1.10.4, 1.11.2, 1.12.0-pre packages using the new "Base" theme (based on 4371d59abd4a24275e0aaa4cae49a2fed591c9ae). They're available for download on https://gist.github.com/rxaviers/5293bc717f283f5eaad7

**Comment by rxaviers** [comment]
> On 1.9 and 1.10, accordion is not working properly in index.html demo page.

Fixed this regression with dc11c77.

**Comment by jzaefferer** [comment]
So with Jasper's last update and this fix, is everything addressed and we can merge both PRs to master? Or are we still missing something?

**Comment by rxaviers** [comment]
It'd be good if @jaspermdegroot could make a final review (packages at https://gist.github.com/rxaviers/5293bc717f283f5eaad7). At least, browsing `index.html` and visualizing if the widgets with new Base theme are correct.

**Comment by rxaviers** [comment]
To Whom It May Concern,

Based on the "we shouldn't bring back those comments/variables" decision (https://github.com/jquery/jquery-ui/pull/1436), TR is going to have dead controls for such variables until the baseline of the available releases for download is at least 1.12.0, where we could remove those controls from TR.

**Comment by jzaefferer** [comment]
> TR is going to have dead controls for such variables until the baseline of the available releases for download is at least 1.12.0, when those controls could be removed from TR.

I'm okay with that. Specifically the shadow controls have probably seen very little use anyway.

Are we ready to land this? I suggest keeping the extra changes until we merge this and the UI theme PR to master, test on stage, then remove it until we release 1.12.

**Comment by rxaviers** [comment]
> Are we ready to land this?

Yeap. I still suggest that @jaspermdegroot make a final review on all packages at https://gist.github.com/rxaviers/5293bc717f283f5eaad7.

> I suggest keeping the extra changes until we merge this and the UI theme PR to master, test on stage, then remove it until we release 1.12.

Ok.

**Comment by jaspermdegroot** [comment]
I suppose the reason that all the changes in the default theme (except changes in ```ui-widget-shadow```) have been applied to previous versions of UI is that the ThemeRoller isn't version aware? Feels a bit wrong though to make changes in something that already has been released. The default theme in the ThemeRoller will be different from the one that you can download via the "Quick Downloads" links at the top of the Download Builder page.

1.12: The color for  ```ui-widget-shadow``` should be ```#666666``` instead of ```#dddddd```. I changed that in my last commit. See https://github.com/jquery/jquery-ui/commit/4371d59abd4a24275e0aaa4cae49a2fed591c9ae

Other than that it looks good!

**Comment by jzaefferer** [comment]
> I suppose the reason that all the changes in the default theme (except changes in ui-widget-shadow) have been applied to previous versions of UI is that the ThemeRoller isn't version aware

That's wrong. Based on the discussion we had around this, it sounded like TR would use the `theme.css` file from the respective release, not just the newest. Was that wrong?

**Comment by rxaviers** [comment]
I'm confused. I think @jaspermdegroot and @jzaefferer are talking about two different things. Correct me if I'm wrong.

I think @jaspermdegroot is talking about the fact the **Custom** Downloads for 1.12.0 and all the previous releases (1.11.2, 1.10.4 and 1.9.2) will include the **Base** theme by default from now on as decided prior to the creation of this PR. Although, the **Quick** Downloads will include **Base** for 1.12.0, but **UI Lightness** for the previous ones (given the fact those are existing old packages). I think Jasper has a valid point. But, the decision is up to you all.

@jzaefferer is talking about either of these two things: (a) the custom downloads should use the `theme.css` file from the respective release, which is true (please, just let me know if it's not); or (b)  the TR (the page, the web interface) should use the `theme.css` file from the respective release, which doesn't happen (TR, the interface, is not release-aware and it uses the latest released jQuery UI's `theme.css`).

**Comment by jzaefferer** [comment]
Alright, I misunderstood that. Let's go ahead and merge this and the UI PR.

**Comment by jzaefferer** [comment]
Upstream change landed in https://github.com/jquery/jquery-ui/commit/79c4fa1e7a09a3d28f0f8ff7a5b9f41a6d6e123d

**Comment by jzaefferer** [comment]
Still need to integrate this into `jqueryui.com` to actually be able to test on stage.
